src/app/qgsmergeattributesdialog.cpp

```diff
@@ -233,7 +233,13 @@ void QgsMergeAttributesDialog::createTableWidgetContents()
   {
     int idx = mTableWidget->horizontalHeaderItem( j )->data( FieldIndex ).toInt();
     bool setToManual = false;
-    if ( !mVectorLayer->dataProvider()->defaultValueClause( idx ).isEmpty() )
+
+    // For the Id use the target feature
+    if ( j == 0 )
```

Comment:
Should this be based on the pk attribute indexes instead? https://github.com/domi4484/QGIS/blob/fixmergeid/src/app/qgsmergeattributesdialog.cpp#L183-L184